app/jobs/benchmark_regression_job.rb

```diff
@@ -0,0 +1,57 @@
+require 'uri'
+require 'net/http'
+require 'json'
+
+class BenchmarkRegressionJob < ActiveJob::Base
+  queue_as :default
+
+  def perform(benchmark_run_id)
+
+    benchmark_run = BenchmarkRun.find_by_id(benchmark_run_id)
+
+    categories = benchmark_run.result.keys
+
+    current_commit = benchmark_run.initiator
+    previous_commits = Commit.where('created_at < ?', current_commit.created_at).where(repo: current_commit.repo).limit(1000)
+
+    previous_benchmark_runs = []
+    previous_commits.each do |commit|
+      previous_benchmark_runs << BenchmarkRun.find_by(initiator: commit, benchmark_type: benchmark_run.benchmark_type,
+                       benchmark_result_type: benchmark_run.benchmark_result_type)
+    end
+
+    categories.each do |category|
+      previous_benchmark_results = previous_benchmark_runs.map { |run| run.result[category].to_f }
+
+      results_param = average(previous_benchmark_results) + 2*standard_deviation(previous_benchmark_results)
+
+      if benchmark_run.result[category].to_f > results_param
+        create_issue benchmark_run, category
+      end
+    end
+  end
+
+  def create_issue(benchmark_run, category)
+    puts "ENV: #{ENV['GITHUB_API']}"
+    puts ENV.to_h
+    uri = URI.parse(Rails.application.secrets.github_api)
+    http_client = Net::HTTP.new(uri.host, uri.port)
+    http_client.use_ssl = true
+    request = Net::HTTP::Post.new(uri.path)
+    request.basic_auth 'RubyBenchBot', Rails.application.secrets.github_password
+    request.body = {"title" => "Benchmark Regression Detected",
+    "body" => "This benchmark regression is caused due to commit #{benchmark_run.initiator.sha1} from the repository #{benchmark_run.initiator.repo.name} for the benchmark #{category}"}.to_json
```

Comment:
https://github.com/ruby-bench/github-api-sandbox/issues/1

I had a look at the above and the details are not meaningful in anyway. Now we know the repo and the commit. But we don't know which `benchmark_type` and for which `benchmark_result_type`. Plus ,  we should probably include how much it has diverted from the standard deviation